snuba/cli/migrations.py

```diff
@@ -39,3 +40,30 @@ def list() -> None:
             click.echo(f"[{symbol}]  {migration_id}{in_progress_text}{blocking_text}")
 
         click.echo()
+
+
+@migrations.command()
+@click.option("--group", required=True, help="Migration group")
+@click.option("--migration-id", required=True, help="Migration ID")
+@click.option("--force", is_flag=True)
+def run(group: str, migration_id: str, force: bool) -> None:
+    """
+    Runs a single migration.
+    The --force option must be passed in order to run blocking migrations.
+
+    Migrations that are already in an in-progress or completed status will not be run.
+    """
+    runner = Runner()
+    migration_key = MigrationKey(MigrationGroup(group), migration_id)
+
+    # If the migration is already completed or underway, do not rerun it
+    status = runner.get_migration_status(migration_key)
+    if status != Status.NOT_STARTED:
+        status_text = status.value
+        click.echo(f"The migration is already {status_text}")
+        return
+
+    runner.run_migration(migration_key, force=force)
```

Comment:
Isn't this going to run any migration by id as long as the status is NOT_STARTED? Which means it could run a migrations independently on the status of the previous migration in the list?

Reply:
Ok, you're right. This might result in some weird states. I changed this to ensure at least that all prior migrations in the group have been completed before the subsequent one is run.